src/lib/formatter/SarifFormatter.ts

```diff
@@ -42,16 +43,16 @@ abstract class SarifFormatter {
 	 * Converts the rule violation to either "warning" or "error"
 	 * See https://docs.oasis-open.org/sarif/sarif/v2.0/csprd02/sarif-v2.0-csprd02.html#_Toc10127839
 	 */
-	protected abstract getLevel(violation: RuleViolation): string;
+	protected abstract getLevel(violation: RuleViolation): Notification.level;
```

Comment:
`Notification.level` is an alias for the possible error types, which align perfectly with what we were returning anyway here.